# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: toolkit/components/antitracking/AntiTrackingIPCUtils.h
# Commit: 312b4abacc2d
# Full Hash: 312b4abacc2dfe0ab2014d62ca271ac54af6704d
# Author: Dimi Lee <dlee@mozilla.com>
# Date: 2020-04-23 21:43:09
# Regressor Bug: 1616775
# File Overlap Count: 1
# Description:
#   Bug 1616775 - P7. Add IPC for ContentBlockingNotifier::OnDecision r=timhuang,baku
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D71016
# ==============================================================================

diff -r 9be5ae499d84 -r 312b4abacc2d toolkit/components/antitracking/AntiTrackingIPCUtils.h
--- a/toolkit/components/antitracking/AntiTrackingIPCUtils.h	Thu Apr 23 14:25:31 2020 +0000
+++ b/toolkit/components/antitracking/AntiTrackingIPCUtils.h	Thu Apr 23 14:25:33 2020 +0000
@@ -25,6 +25,14 @@
           mozilla::ContentBlockingNotifier::StorageAccessGrantedReason::
               eOpener> {};
 
+// ContentBlockingNotifier::BlockingDecision over IPC.
+template <>
+struct ParamTraits<mozilla::ContentBlockingNotifier::BlockingDecision>
+    : public ContiguousEnumSerializerInclusive<
+          mozilla::ContentBlockingNotifier::BlockingDecision,
+          mozilla::ContentBlockingNotifier::BlockingDecision::eBlock,
+          mozilla::ContentBlockingNotifier::BlockingDecision::eAllow> {};
+
 // ContentBlocking::StorageAccessPromptChoices over IPC.
 template <>
 struct ParamTraits<mozilla::ContentBlocking::StorageAccessPromptChoices>